sample_code/ep319/README.md

```diff
@@ -0,0 +1,63 @@
+# [git: main vs. origin/main (beginner - intermediate)](https://youtu.be/f-92I_gxgjA)
+
+Today I go over the difference between `main` and `origin/main` as well as the most reliable ways to get a fresh branch and to pull in fresh changes from a git remote!
+
+## Setup commands
+
+Simulate clone of git repository back in time:
+
+```bash
+# git clone --bare git@github.com:asottile/astpretty
+git clone --bare https://github.com/asottile/astpretty
+
+git clone astpretty.git/ astpretty
+cd astpretty
+
+git reset --hard 8455ab7f
+git push --force origin master
+
+# git remote set-url origin git@github.com:asottile/astpretty
+git remote set-url origin https://github.com/asottile/astpretty
+```
```

Comment:
Attempt for simulating your setup from the video -- not sure if there is a shorter way of doing it (without `bare` repo), aka with shorter commands.

Reply:
ah yeah I think what I did was:

```bash
git clone git@github.com:asottile/astpretty tmp
git -C tmp reset --hard HEAD^^^^
git clone tmp astpretty
git -C astpretty remote set-url git@github.com:asottile/astpretty
```